src/ApiManagement/ApiManagement.ServiceManagement/Commands/GetAzureApiManagementSubscription.cs

```diff
@@ -43,20 +44,37 @@ public class GetAzureApiManagementSubscription : AzureApiManagementCmdletBase
             HelpMessage = "Subscription identifier. If specified will try to find subscription by the identifier. This parameter is optional.")]
         public String SubscriptionId { get; set; }
 
+        [Parameter(
+            ParameterSetName = GetByProductIdAndUser,
+            ValueFromPipelineByPropertyName = true,
+            Mandatory = true,
+            HelpMessage = "User identifier. If specified will try to find all subscriptions by the user identifier. This parameter is optional.")]
         [Parameter(
             ParameterSetName = GetByUserId,
             ValueFromPipelineByPropertyName = true,
             Mandatory = false,
             HelpMessage = "User identifier. If specified will try to find all subscriptions by the user identifier. This parameter is optional.")]
         public String UserId { get; set; }
 
+        [Parameter(
+            ParameterSetName = GetByProductIdAndUser,
+            ValueFromPipelineByPropertyName = true,
+            Mandatory = true,
+            HelpMessage = "Product identifier. If specified will try to find all subscriptions by the product identifier. This parameter is optional.")]
         [Parameter(
             ParameterSetName = GetByProductId,
             ValueFromPipelineByPropertyName = true,
             Mandatory = false,
             HelpMessage = "Product identifier. If specified will try to find all subscriptions by the product identifier. This parameter is optional.")]
         public String ProductId { get; set; }
 
+        [Parameter(
+            ParameterSetName = GetByScope,
+            ValueFromPipelineByPropertyName = true,
+            Mandatory = false,
```

Comment:
@solankisamir this parameter should be mandatory to avoid any possible conflicts with the `GetAllSubscriptions` parameter set -- you'll also want to regenerate the help for this cmdlet after making this change

Reply:
fixed